$NetBSD: patch-ab,v 1.6 2024/11/04 16:06:12 nia Exp $

* YACC and LEX are passwd as full path.
* static linking is not supported on MacOS X.  PR#39058.
* let pkgsrc take care of hardening flags
* FILE_OFFSET_BITS corrected

--- configure.orig	2010-09-10 17:48:04.000000000 +0000
+++ configure
@@ -4231,10 +4231,6 @@ fi
 done
 test -n "$YACC" || YACC="yacc"
 
-if test "x${YACC}" != "xbison -y"; then
-        echo "AIDE requires GNU bison"
-        exit 5
-fi
 
 
 for ac_prog in flex lex
@@ -4394,10 +4390,6 @@ fi
 rm -f conftest.l $LEX_OUTPUT_ROOT.c
 
 fi
-if test "x${LEX}" != "xflex"; then
-        echo "AIDE requires GNU flex"
-        exit 5
-fi
 for ac_prog in ld
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
@@ -4539,6 +4531,9 @@ if test $ac_cv_c_compiler_gnu = yes; the
 	## pic_flag=-Kconform_pic
 	## fi
 	;;
+    darwin*)
+	LD_STATIC_FLAG=''
+	;;
     *)
 	## pic_flag='-fPIC'
 	;;
@@ -4694,16 +4689,6 @@ if test "$aide_static_choice" != "yes"; 
   EXTRA_LDFLAGS=""
   EXTRA_CFLAGS=""
   if test x$CC = "xgcc"; then
-    if ld_supports_flag -z,relro; then
-      EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-z,relro"
-    fi
-    if ld_supports_flag -z,now; then
-      EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-z,now"
-    fi
-    if cc_supports_flag -fPIE -DPIE; then
-      EXTRA_CFLAGS="$EXTRA_CFLAGS -fPIE -DPIE"
-      EXTRA_LDFLAGS="$EXTRA_LDFLAGS -pie"
-    fi
         if cc_supports_flag -Wundef; then
       EXTRA_CFLAGS="$EXTRA_CFLAGS -Wundef"
     fi
@@ -6435,7 +6420,7 @@ ac_fn_c_check_func "$LINENO" "lstat64" "
 if test "x$ac_cv_func_lstat64" = x""yes; then :
   ac_fn_c_check_func "$LINENO" "stat64" "ac_cv_func_stat64"
 if test "x$ac_cv_func_stat64" = x""yes; then :
-  AIDE_DEFS="$AIDE_DEFS -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS"
+  AIDE_DEFS="$AIDE_DEFS -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64"
 	AIDE_LSTAT_FUNC="lstat64"
 	compoptionstring="${compoptionstring}WITH_LSTAT64\\n"
 	AIDE_STAT_FUNC="stat64"
